aiohttp/streams.py

```diff
@@ -84,7 +90,7 @@ def iter_chunks(self) -> ChunkTupleAsyncStreamIterator:
         return ChunkTupleAsyncStreamIterator(self)  # type: ignore[arg-type]
 
 
-class StreamReader(AsyncStreamReaderMixin):
+class StreamReader(AsyncStreamReaderMixin, ErrorableMixin):
```

Comment:
I wonder if it makes more sense to use a Protocol, rather than needing to inherit from this mixin (which doesn't actually define any functionality).

Reply:
Dunno. I never worked with the protocols much. Just know they exist and are callback/hook collections. Plz send examples of how you envision this. Currently, this was meant as a quick hack to please MyPy and get the PoC going.

Reply:
```
class Errorable(Protocol):
    def set_exception(
        self,
        exc: BaseException,
        exc_cause: BaseException = ...,
    ):
        ...
```

Then just use that for the parameter type, anything that has a compatible set_exception() method will work then, without needing to subclass anything.

Reply:
```suggestion
class StreamReader(AsyncStreamReaderMixin):
```